tv_apps/tv-epg/js/epg_program.js

```diff
@@ -0,0 +1,74 @@
+/* global EPG */
+
+'use strict';
+
+window.EPGProgram = (function(exports) {
+  var proto = Object.create(HTMLElement.prototype);
+
+  proto.createdCallback = function() {
+    this.duration = 1;
+    this.dataset.duration = '1';
+    this.titleElement = document.createElement('DIV');
+    this.titleElement.classList.add('title');
+    this.appendChild(this.titleElement);
+
+    this.progressElement = document.createElement('DIV');
+    this.progressElement.classList.add('background-progress');
+    this.appendChild(this.progressElement);
+  };
+
+  proto.resetProgressElement = function(time) {
+    if (!this.startTime || !time) {
+      return;
+    }
+
+    if (this.startTime + this.duration <= time) {
+      this.fillProgress();
+    } else {
+      this.progressElement.classList.add('smooth');
+    }
+    this.appendChild(this.progressElement);
+  };
+
+  proto.setProgress = function(time) {
+    if (!this.startTime || !time) {
+      return;
+    }
+
+    var scaleX = (time - this.startTime) * EPG.COLUMN_WIDTH;
+    scaleX = scaleX / (this.duration * EPG.COLUMN_WIDTH - EPG.COLUMN_MARGIN);
+    scaleX = Math.min(scaleX, 1);
+    this.progressElement.style.transform = 'scaleX(' + scaleX + ')';
+  };
+
+  proto.fillProgress = function() {
+    this.progressElement.style.transform = 'scaleX(1)';
+  };
+
+  proto.setStartTime = function(time) {
```

Comment:
How about to use property getter and setter for these parts?

`startTime, title, titlePadding (maybe), duration`